play-json/jvm/src/main/scala/play/api/libs/json/EnvReads.scala

```diff
@@ -415,6 +419,78 @@ trait EnvReads {
     case _ => JsError(JsonValidationError("error.expected.jsstring"))
   }
 
+  private def javaDurationNumberReads(unit: TemporalUnit, esuffix: String) =
+    Reads[JDuration] {
+      case JsNumber(n) if !n.ulp.isWhole =>
+        JsError(s"error.invalid.$esuffix")
```

Comment:
Also, you probably don't need the `esuffix` anymore since this is only used for `Duration`.